File: src/ExpenseManagement.Domain/Entities/Subscription.cs
using ExpenseManagement.Domain.Common;

namespace ExpenseManagement.Domain.Entities;

public class Subscription : AuditableEntity
{
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public decimal Amount { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime? EndDate { get; set; }
    public DateTime NextBillingDate { get; set; }
    public int BillingCycle { get; set; } // Days between billings
    public bool IsActive { get; set; }
    public int ReminderDaysBefore { get; set; } = 3;
    public string UserId { get; set; } = string.Empty;
    public Guid CategoryId { get; set; }
    public Guid? CurrencyId { get; set; }

    // Navigation properties
    public Category Category { get; set; } = null!;
    public Currency? Currency { get; set; }
}

File: expense-management-ui/src/types/subscription.ts
export interface Subscription {
  id: string;
  name: string;
  description?: string;
  amount: number;
  startDate: string;
  endDate?: string;
  nextBillingDate: string;
  billingCycle: number;
  isActive: boolean;
  reminderDaysBefore: number;
  categoryId: string;
  categoryName: string;
  categoryColor: string;
  currencyId?: string;
  currencyCode?: string;
  currencySymbol?: string;
}

export interface CreateSubscriptionDto {
  name: string;
  description?: string;
  amount: number;
  startDate: string;
  endDate?: string;
  billingCycle: number;
  reminderDaysBefore: number;
  categoryId: string;
  currencyId?: string;
}
